polkadot-protocol/basics/data-encoding.md

```diff
@@ -1,4 +1,105 @@
 ---
 title: Data Encoding
-description: TODO
----
\ No newline at end of file
+description: SCALE codec enables fast, efficient data encoding in Substrate, ideal for resource-constrained environments like Wasm, supporting custom types and compact encoding.
+---
+
+# Data Encoding
+
+Polkadot SDK uses a lightweight and efficient encoding/decoding mechanism to optimize data transmission across the network. This mechanism, known as the **SCALE** codec, is used for serializing and deserializing data.
+
+The **SCALE** codec plays a vital role in enabling communication between the runtime and the outer node.
+
+The **SCALE** codec is designed for high-performance, copy-free data encoding and decoding in resource-constrained environments like the Polkadot SDK [WebAssembly runtime](add link to Runtime development page){target=\_blank}.
+
+It is not self-describing, meaning the decoding context must have full knowledge of the encoded data types. 
+
+Parity's front-end libraries utilize the [`parity-scale-codec`](https://github.com/paritytech/parity-scale-codec){target=\_blank} crate (a Rust implementation of the SCALE codec) to handle encoding and decoding for interactions between RPCs and the runtime.
+
+The codec is ideal for Polkadot SDK chains and blockchain systems because:
+
+- It is lightweight compared to generic serialization frameworks like [`serde`](https://serde.rs/){target=\_blank}, which add unnecessary bulk to binaries.
+- It doesn’t rely on Rust’s `libstd`, making it compatible with `no_std` environments like Wasm runtime.
+- It integrates seamlessly with Rust, allowing easy derivation of encoding and decoding logic for new types using `#[derive(Encode, Decode)]`.
+
+Defining a custom encoding scheme in Polkadot SDK chain, rather than using an existing Rust codec library, is crucial for enabling cross-platform and multi-language support. 
+
+By creating the SCALE codec, Substrate ensures that the encoding can be re-implemented in other languages and platforms to support interoperability between Substrate-based blockchains.
+
+## Parity SCALE Codec
+
+Rust implementation of the SCALE data format for types used in the Polkadot SDK framework.
+
+The codec is implemented using the following traits:
+
+- Encode
+- Decode
+- CompactAs
+- HasCompact
+- EncodeLike
+
+### Encode
+
+The `Encode` trait handles data encoding into SCALE format and includes the following key functions:
+
+- s`ize_hint(&self) -> usize`: Estimates the number of bytes required for encoding to prevent multiple memory allocations. This should be inexpensive and avoid complex operations. Optional if the size isn’t known.
```

Comment:
```suggestion
- `size_hint(&self) -> usize`: Estimates the number of bytes required for encoding to prevent multiple memory allocations. This should be inexpensive and avoid complex operations. Optional if the size isn’t known.
```

Reply:
I'd the a reference to the rust docs here, for example:

- [`size_hint(&self) -> usize`](https://docs.rs/parity-scale-codec/latest/parity_scale_codec/trait.Encode.html#method.size_hint){target=\_blank}: Estimates the number of bytes required for encoding to prevent multiple memory allocations. This should be inexpensive and avoid complex operations. Optional if the size isn’t known.

And the same for the rest of the functions